MicroBenchmarks/LoopVectorization/LoopInterleaving.cpp

```diff
@@ -0,0 +1,313 @@
+
+#include "benchmark/benchmark.h"
+
+#define ELEMENTS 2048
+#define ALIGNED16 __attribute__((aligned(16)))
+
+static std::mt19937 rng;
+unsigned int g_sum = 0;
+
+int A[ELEMENTS] ALIGNED16;
+int B[ELEMENTS] ALIGNED16;
+int C[ELEMENTS] ALIGNED16;
+
+// Initialize arrays with random numbers.
+static void init_data(unsigned N) {
+  std::uniform_int_distribution<int> distrib(std::numeric_limits<int>::min(),
+                                             std::numeric_limits<int>::max());
+  for (unsigned I = 0; I < N; I++) {
+    A[I] = distrib(rng);
+    B[I] = distrib(rng);
+    C[I] = distrib(rng);
+  }
+}
+
+static void __attribute__((noinline)) loopWithVW4IC1(int Iterations) {
+#pragma clang loop vectorize_width(4) interleave(disable)
+  for (int J = 0; J < Iterations; J++) {
+    A[J] = B[J] + C[J];
+  }
+}
+
+static void __attribute__((noinline)) loopWithVW4IC2(int Iterations) {
+#pragma clang loop vectorize_width(4) interleave_count(2)
+  for (int J = 0; J < Iterations; J++) {
+    A[J] = B[J] + C[J];
+  }
+}
+
+static void __attribute__((noinline)) loopWithVW4IC4(int Iterations) {
+#pragma clang loop vectorize_width(4) interleave_count(4)
+  for (int J = 0; J < Iterations; J++) {
+    A[J] = B[J] + C[J];
+  }
+}
+
+static int __attribute__((noinline))
+loopWithReductionWithVW4IC1(int Iterations) {
+  unsigned sum = 0;
+#pragma clang loop vectorize_width(4) interleave(disable)
+  for (int J = 0; J < Iterations; J++) {
+    sum += A[J];
+  }
+  return sum;
+}
+
+static int __attribute__((noinline))
+loopWithReductionWithVW4IC2(int Iterations) {
+  unsigned sum = 0;
+#pragma clang loop vectorize_width(4) interleave_count(2)
+  for (int J = 0; J < Iterations; J++) {
+    sum += A[J];
+  }
+  return sum;
+}
+
+static int __attribute__((noinline))
+loopWithReductionWithVW4IC4(int Iterations) {
+  unsigned sum = 0;
+#pragma clang loop vectorize_width(4) interleave_count(4)
+  for (int J = 0; J < Iterations; J++) {
+    sum += A[J];
+  }
+  return sum;
+}
+
+static int __attribute__((noinline))
+loopWithReductionWithVW1IC1(int Iterations) {
+  unsigned sum = 0;
+#pragma clang loop vectorize_width(1) interleave_count(1)
+  for (int J = 0; J < Iterations; J++) {
+    sum += A[J];
+  }
+  return sum;
+}
+
+static int __attribute__((noinline))
+loopWithReductionWithVW1IC2(int Iterations) {
+  unsigned sum = 0;
+#pragma clang loop vectorize_width(1) interleave_count(2)
+  for (int J = 0; J < Iterations; J++) {
+    sum += A[J];
+  }
+  return sum;
+}
+
+static int __attribute__((noinline))
+loopWithReductionWithVW1IC4(int Iterations) {
+  unsigned sum = 0;
+#pragma clang loop vectorize_width(1) interleave_count(4)
+  for (int J = 0; J < Iterations; J++) {
+    sum += A[J];
+  }
+  return sum;
+}
+
+static void __attribute__((always_inline))
+benchForLoopInterleaveThreshold(benchmark::State &state, void (*Fn)(int),
+                                int Iterations) {
+  std::uniform_int_distribution<int> distrib(std::numeric_limits<int>::min(),
```

Comment:
unused?

Reply:
Merged the functionality in one function & deleted this copy.